flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/relational/connection/JdbcConnectionPools.java

```diff
@@ -31,16 +33,16 @@ public class JdbcConnectionPools implements ConnectionPools<HikariDataSource, Jd
 
     private static JdbcConnectionPools instance;
     private final Map<ConnectionPoolId, HikariDataSource> pools = new HashMap<>();
-    private static JdbcConnectionPoolFactory jdbcConnectionPoolFactory;
+    private static Map<String, JdbcConnectionPoolFactory> poolFactorys = new HashMap<>();
 
     private JdbcConnectionPools() {}
 
     public static synchronized JdbcConnectionPools getInstance(
             JdbcConnectionPoolFactory jdbcConnectionPoolFactory) {
         if (instance == null) {
-            JdbcConnectionPools.jdbcConnectionPoolFactory = jdbcConnectionPoolFactory;
             instance = new JdbcConnectionPools();
         }
+        poolFactorys.put(jdbcConnectionPoolFactory.getDriverClassName(), jdbcConnectionPoolFactory);
```

Comment:
Since one driver class may be used for multiple kinds of databases, if this factory class adds other configurations other than jdbcUrl in the future, there may be no way to distinguish them by the driver class name. So I think it would be better to use a kind of type identifier here.

Reply:
I have been thought type identifier at first, but I have no idea that how to get this identifier from mysql.source.connection.JdbcConnectionFactory#connect .The param is JdbcConfiguration, so I read a unique data(driver name) from it.

If use type identifier, I think also a param identifier should add to JdbcConnectionFactory. WDYT, @whhe

Reply:
Or maybe take the factory class name as the map key?

Reply:
Or take the factory class name as the map key?

Reply:
@whhe , I have modify it, CC.